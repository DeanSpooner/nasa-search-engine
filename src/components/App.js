import React, { useState } from "react";
import Search from "./Search";
import SearchResults from "./SearchResults";
import "../styles/App.css";

const App = () => {
  const [searchResults, setSearchResults] = useState([]);

  return (
    <div className="App">
      <img
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="NASA logo"
        className="nasa_image"
      ></img>
      <Search setSearchResults={setSearchResults} />
      <SearchResults results={searchResults} />
    </div>
  );
};

export default App;
