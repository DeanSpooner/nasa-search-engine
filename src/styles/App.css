.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.nasa_image {
  margin: auto;
  width: 20%;
}
