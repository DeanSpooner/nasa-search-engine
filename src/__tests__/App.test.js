import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

test("renders NASA logo on screen", () => {
  render(<App />);
  const nasaLogo = screen.getByAltText(/NASA logo/i);
  expect(nasaLogo).toBeInTheDocument();
});
