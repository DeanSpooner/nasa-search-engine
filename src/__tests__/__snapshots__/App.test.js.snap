// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="NASA logo"
      class="nasa_image"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <form
      class="search__form"
    >
      <input
        class="search__input"
        data-testid="search__input"
        type="text"
      />
      <button
        class="search__button"
        data-testid="search__button"
        type="submit"
      >
        Search
      </button>
    </form>
    <p>
      No results
    </p>
  </div>
</DocumentFragment>
`;
